### PR TITLE
Anthony McKeever - Lesson 4 Assignment and Exercise Submission

### PR DESCRIPTION
**Pull request by Snip3rM00n**
This is my submission for lesson 4:
- Assignment - Mailroom Part 2
- Exercise 1: Dictionary Lab
- Exercise 2: Files Lab
- Exercise 3: Trigrams

I've also included the files I used for testing the files labs and trigrams lab.  If these are unnecessary I'd be happy to remove them from this PR.  Just let me know. 🙂 

---

_**Squashed Commit History:**_

* Initial commit of lesson 4 exercises

* bug fix

* Fix bugs

* Mailroom Part 2 initial commit

* Optimization and bug fixes

* Finalize assignment

Refactor code to use dictionaries
Implement menu system
Implement send letters to all
Implement file writting
Document all the things!

* Final commit of lesson 4 work.

Adds trigrams for days.
Adds test files.
Change test image - test_jpg.jpg to a smaller file.
updates readme's source map

* Fix white space

* Remove ending line tab

* ditto last commit

### CODE REVIEW

File: students/anthony_mckeever/lesson4/assignment_1/mailroom.py [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+              "Delilah Matsuka":     ([199.99, 299.99, 2100.00],         get_donor),
+              "Astra Matsume":       ([599.99],                          get_donor),
+              "Kima Metoyo":         ([3600.00, 1200.00],                get_donor),
+              "Kayomi Matsuka":      ([0.01],                            get_donor),
+              "Katie Starchelle":    ([600.00],                          get_donor)
+              }
+
+
+if __name__ == "__main__":
+    main()
```
Comment (charlesrobison) [inline students/anthony_mckeever/lesson4/assignment_1/mailroom.py]: This works really well and I appreciate how thorough you're being!

**Comment by Snip3rM00n** [comment]
> Not able to run trigrams - is it ready for review? If not, please submit a new PR when ready.

Thanks for the sign off, @charlesrobison !  :)  I'm surprised trigrams couldn't run, is there a particular issue you were running into?  I'd be glad to make sure it gets fixed and submit a PR for it.

**Comment by charlesrobison** [comment]
@Snip3rM00n for trigrams, I was running it incorrectly - figured it out.  Works great - nice job!  Did you submit it on the course site for grading?

**Comment by Snip3rM00n** [comment]
> @Snip3rM00n for trigrams, I was running it incorrectly - figured it out.  Works great - nice job!  Did you submit it on the course site for grading?

@charlesrobison - Glad it worked for you!  I totally forgot to submit the file through the course site.  I've submitted it now. :)

**Pull request merged**